### PR TITLE
fix: adjust spin styles to support sm/lg variants

### PR DESCRIPTION
**Pull request by sribich**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

#38922 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Fix Spin alignment when using `tip`. |
| 🇨🇳 Chinese | 修复 Spin 包裹模式时的样式偏移问题。         |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by sribich** [comment]
Since this is a style only change, this cannot be tested using normal component tests.

I also don't think it's possible to add an image test either due to the animated nature of the component. I would test to see if this is in fact the case, but unfortunately `npm run test-image` is failing for me.

Do you have any recommendations for adding tests, or is this an instance it's okay to skip?

**Comment by sribich** [comment]
Also is adding a native Chinese translation something a maintainer will do, or how should I go about adding those? 
I would prefer to not add a low quality machine translation, but if that's the norm then I can do that.

**Comment by sribich** [comment]
I have updated the docs to show sized tip examples:

image

File: components/spin/style/index.tsx [hunk 1]
```diff
@@ -71,7 +71,7 @@ const genSpinStyle: GenerateStyle<SpinToken> = (token: SpinToken): CSSObject =>
           marginTop: -(token.spinDotSize / 2) - 10,
         },
 
-        [`> div > ${token.componentCls}-sm`]: {
+        '&-sm': {
```
Comment (afc163) [inline components/spin/style/index.tsx:74]: Why update this?

Comment (sribich) [reply components/spin/style/index.tsx:74]: Removing all of the css rules to illustrate the problem gives us this:

```typescript
const genSpinStyle: GenerateStyle<SpinToken> = (token: SpinToken): CSSObject => ({
  [`${token.componentCls}`]: {
    '&-nested-loading': {
      [`> div > ${token.componentCls}`]: {
        [`> div > ${token.componentCls}-sm`]: {},
        [`> div > ${token.componentCls}-lg`]: {},
      },
    },
  },
});
```

## Proposed solution 1

```typescript
const genSpinStyle: GenerateStyle<SpinToken> = (token: SpinToken): CSSObject => ({
  [`${token.componentCls}`]: {
    '&-nested-loading': {
      [`> div > ${token.componentCls}`]: {
        '&-sm': {},
        '&-lg': {},
      },
    },
  },
});
```

## Proposed solution 2

```typescript
const genSpinStyle: GenerateStyle<SpinToken> = (token: SpinToken): CSSObject => ({
  [`${token.componentCls}`]: {
    '&-nested-loading': {
      [`> div > ${token.componentCls}`]: {},
      [`> div > ${token.componentCls}-sm`]: {},
      [`> div > ${token.componentCls}-lg`]: {},
    },
  },
});
```

**Comment by afc163** [comment]
Grate fix, we need one last step to merge, could you run `npm test -- components/spin -u` in local and commit snapshot update?

**Comment by sribich** [comment]
Snapshots have been updated

**Review by afc163 — APPROVED** [review]

**Pull request merged**